### ISSUE TITLE
Prepare release v.1.9.0 (tentative: 3rd March)

### ISSUE THREAD

**Issue by lalitb** [open-telemetry/opentelemetry-cpp#2007]
There are few Open PRs which preferably be merged (not in order of priority) before creating the release:

* #2005 
* #2004 
* #2000 
* #1993 
* #1909 

#2000 is  ABI breaking change for the SDK which should be fine in general. But it also means that if someone is using otel-cpp-sdk*so targets with their application, the application can crash if these shared library is rebuild for new version, without recompiling/relinking the application against these new shared libraries. This needs to be mentioned in the release notes.

**Comment by marcalff** [comment]
Please revert the fix done with PR: Rename the global SDK version variables to avoid naming clash #2011 .

OPENTELEMETRY_SDK_MAJOR_VERSION needs to be a pre processor symbol to fix issue  Provide major/minor/patch version macros #2012 .

**Comment by marcalff** [comment]
@lalitb @ThomsonTan @esigo Please see above comment.

**Comment by ThomsonTan** [comment]
> Please revert the fix done with PR: Rename the global SDK version variables to avoid naming clash #2011 .
> 
> OPENTELEMETRY_SDK_MAJOR_VERSION needs to be a pre processor symbol to fix issue Provide major/minor/patch version macros #2012 .

Is the revert required? I think the PR #2011 is separate the the ask of #2012 which looks like a feature request? The PR was targeting a different issue of name clashing.

**Comment by owent** [comment]
I found some commits are missed in #2005,Can I push them next monday?I added a CI job and a unit test to check #1603.

**Comment by ThomsonTan** [comment]
> I found some commits are missed in #2005,Can I push them next Monday added a CI job and a unit test to check #1603.

It looks good to me. We can wait it before completing the release. @lalitb @marcalff @esigo how do you think?

**Comment by marcalff** [comment]
> > Please revert the fix done with PR: Rename the global SDK version variables to avoid naming clash #2011 .
> > OPENTELEMETRY_SDK_MAJOR_VERSION needs to be a pre processor symbol to fix issue Provide major/minor/patch version macros #2012 .
> 
> Is the revert required? I think the PR #2011 is separate the the ask of #2012 which looks like a feature request? The PR was targeting a different issue of name clashing.

I think the revert is required.

Code introduced:

```
namespace sdk
{
namespace version
{
extern const int OPENTELEMETRY_SDK_MAJOR_VERSION;
```

will break once pre processor symbol is added for #2012 :

```
#define OPENTELEMETRY_SDK_MAJOR_VERSION 1
```

**Comment by lalitb** [comment]
The intended purpose of the existing const variables was to stamp the version information in the binary.

```
$strings /usr/local/lib/libopentelemetry_version.a  | grep 1.8.2
pre_release_1.8.2
1.8.2
1.8.2-NONE-NONE-37-pre_release_1.8.2-435ce60f233b6718aaa04bb7068dd641b536299b
```

We shouldn't be removing them, or replacing them with macros. Though it should be fine to change these variable names to lowercase (or something else) for supporting #2012.

**Comment by marcalff** [comment]
I agree these const variables should be available in the library, this is not the problem.

The problem is that resolving one name clash for MAJOR_VERSION created another name slash instead.

Ok to change to lowercase instead:

```
namespace sdk
{
namespace version
{
extern const int major_version;
```

This avoid the conflict with MAJOR_VERSION, and with OPENTELEMETRY_SDK_MAJOR_VERSION anticipated for #2012

**Comment by marcalff** [comment]
@ThomsonTan @lalitb 

So, if `sdk::version::major_version` looks ok to everyone, no need for a full revert, but some adjustments are needed instead before publishing the release.

**Comment by ThomsonTan** [comment]
Ok. I'll make a separate PR to change the C++ version variables into lower case.

**Comment by marcalff** [comment]
@lalitb About the version name, why "1.9.2" instead of "1.9.0" ?

**Comment by lalitb** [comment]
> 
@lalitb About the version name, why "1.9.2" instead of "1.9.0" ?

**Comment by owent** [comment]
#2005 and #2000 are all finished now.

**Comment by marcalff** [comment]
Can we also add this to the next release:

* https://github.com/open-telemetry/opentelemetry-cpp/issues/2016

I will generate new semconv once the spec label is ready.
